packager/packager.cc

```diff
@@ -963,13 +970,39 @@ struct PackageInfoFinder {
                 info->importedPackageNames.emplace_back(move(name), method2ImportType(send));
             }
         }
+
         if (send.fun == core::Names::restrict_to_service() && send.numPosArgs() == 1) {
             // Transform: `restrict_to_service Foo` -> `restrict_to_service <PackageSpecRegistry>::Foo`
             auto importArg = move(send.getPosArg(0));
             send.removePosArg(0);
             ENFORCE(send.numPosArgs() == 0);
             send.addPosArg(prependName(move(importArg), core::Names::Constants::PackageSpecRegistry()));
         }
+
+        if (send.fun == core::Names::autoloader_compatibility() && send.numPosArgs() == 1) {
+            // Parse autoloader_compatibility DSL and set strict bit on PackageInfoImpl if configured
+            auto *compatibilityAnnotationLit = ast::cast_tree<ast::Literal>(send.getPosArg(0));
+            if (compatibilityAnnotationLit == nullptr || !compatibilityAnnotationLit->isString()) {
+                if (auto e = ctx.beginError(send.loc, core::errors::Packager::InvalidConfiguration)) {
+                    e.setHeader("Argument to `{}` must be a string literal", send.fun.show(ctx));
```

Comment:
What do you think about adding an autocorrect for the case where the user wrote a symbol but it was a valid argument? `:legacy` -> `'legacy'` etc.

Reply:
I think this could be useful, but would probably want to do this down the line. It's not clear how much our users will be adding annotations here vs our package-generation tooling.